packages/ec-core/src/weighted/weighted_pair.rs

```diff
@@ -0,0 +1,179 @@
+use rand::distr::{Bernoulli, Distribution};
+
+use super::{
+    error::{SelectionError, WeightSumOverflow, WeightedPairError, ZeroWeight},
+    with_weight::WithWeight,
+};
+use crate::{operator::selector::Selector, population::Population};
+
+#[derive(Debug)]
+pub struct WeightedPair<A, B> {
+    pub(crate) a: A,
+    pub(crate) b: B,
+    pub(crate) distr: Option<Bernoulli>,
+    pub(crate) weight_sum: u32,
```

Comment:
Just one thing I noticed, not that we necessarily need to change this, but we should consider it: Do we want the struct items ordered this way?

Reply:
I'd be inclined to keep it this way.